### PR TITLE
Fix j326 peer regexp, and IP parse/validation with asio/boost

### PR DESCRIPTION
**Pull request by rfree-d**

### CODE REVIEW

**Comment by rfree-d** [comment]
wrong branch, opening other instead

File: src/galaxy42_utils.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	std::regex pattern("\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}:\\d{1,5}"); // i.e. 127.0.0.1:4562
+	std::smatch result;
+	if (!std::regex_search(ip_string, result, pattern)) { // bad argument
+//		throw std::invalid_argument("bad format of input ip address");
+                throw std::invalid_argument(boost::locale::gettext("L_bad_format_of_input_ip_address"));
+
+	}
+	*/
+	size_t pos = ip_string.find(':');
+	if (pos == string::npos) _throw_error( runtime_error( join_string_sep("Invalid IP format (char ':')", ip_string)));
```
Comment (robertoleksy) [inline src/galaxy42_utils.cpp]: we should throw std::invalid_argument

File: src/c_ip46_addr.cpp [hunk 2]
```diff
@@ -83,12 +84,13 @@ bool c_ip46_addr::is_ipv4(const string &ipstr) {
...
 	hint.ai_flags = AI_NUMERICHOST;
+
 	int ret = getaddrinfo(ipstr.c_str(), nullptr, &hint, &result);
-	if (ret) {
-		_throw_error( std::invalid_argument("unknown address format") );
-	}
-	auto result_deleter = [&](struct addrinfo *result){freeaddrinfo(result);};
+	if (ret) _throw_error( invalid_argument( join_string_sep("unknown address format, ret",ret,"for ipstr",ipstr)));
+	if (!result) _throw_error( invalid_argument( join_string_sep("unknown address format, pointer result",result,"for ipstr",ipstr)));
+	auto result_deleter = [&](struct addrinfo *result){ if (!result) _throw_error(runtime_error("NULL in freeaddrinfo"); freeaddrinfo(result)); };
```
Comment (robertoleksy) [inline src/c_ip46_addr.cpp]: fixed in another PR

File: src/galaxy42_utils.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+	std::string ip = ip_string.substr(0, pos);
+	if (ip.size()<(1+1)*4) _throw_error(runtime_error(join_string_sep("Invalid (too small) IP size (ip part)",ip.size())));
```
Comment (robertoleksy) [inline src/galaxy42_utils.cpp]: magic number

File: src/galaxy42_utils.hpp [hunk 4]
```diff
@@ -0,0 +1,8 @@
+
+#pragma once
+
+#include <libs0.hpp>
+
+/// Converts string like "100.200.50.50:32000"
```
Comment (robertoleksy) [inline src/galaxy42_utils.hpp]: add ipv4 to function name

**Pull request merged**